File: algorithms/programmers/42586/Q42586.js
function solution(progresses, speeds) {
    var answer = [];
    var days = [];

    for(var i=0; i<progresses.length; i++){
        var q = (100 - progresses[i]) / speeds[i];
		days.push(Math.ceil(q));
    }
    
    var tasks = 0;
    var troll = days[0];
    
    for(var i=1; i<=days.length; i++){
        tasks++;
        if(i == days.length){
            answer.push(tasks);
            break;
        } else if(troll < days[i]) {
            answer.push(tasks);
            tasks = 0;
            troll = days[i];
        }
    }

    return answer;
}

File: algorithms/programmers/42586/Q42586.java
class Solution {
	public Stack<Integer> solution(int[] progresses, int[] speeds) {
		Stack<Integer> result = new Stack<>();
		int[] days = new int[progresses.length];

		// 진도 100% 달성까지 남은 날짜 수 계산
		for (int i = 0; i < progresses.length; i++) {
			int q = (100 - progresses[i]) / speeds[i];
			int r = (100 - progresses[i]) % speeds[i];
			days[i] = q + ((r == 0) ? 0 : 1);
		}

		int tasks = 0;
		int troll = days[0];

		// 기능 배포시점에서 배포할 기능의 개수 계산
		for (int i = 1; i <= days.length; i++) {
			tasks++;
			if (i == days.length) {
				result.push(tasks);
				break;
			} else if (troll < days[i]) {
				result.push(tasks);
				tasks = 0;
				troll = days[i];
			}
		}

		return result;
	}
}